integration-tests/smoke/keeper_test.go

```diff
@@ -638,6 +644,77 @@ var _ = Describe("Keeper Suite @keeper", func() {
 			}, "1m", "1s").Should(Succeed())
 		}
 
+		if testToRun == HandleKeeperNodesGoingDown {
+			By("taking down Keeper nodes has a visible effect on the upkeep servicing")
+			var initialCounters = make([]*big.Int, len(upkeepIDs))
+
+			// Watch upkeeps being performed and store their counters in order to compare them later in the test
+			Eventually(func(g Gomega) {
+				for i := 0; i < len(upkeepIDs); i++ {
+					counter, err := consumers[i].Counter(context.Background())
+					initialCounters[i] = counter
+					g.Expect(err).ShouldNot(HaveOccurred(), "Failed to retrieve consumer counter"+
+						" for upkeep at index "+strconv.Itoa(i))
+					g.Expect(counter.Int64()).Should(BeNumerically(">", int64(0)),
+						"Expected consumer counter to be greater than 0, but got %d", counter.Int64())
+				}
+			}, "1m", "1s").Should(Succeed())
+
+			// Take down half of the Keeper nodes by deleting the Keeper job registered above (after registry deployment)
+			firstHalfToTakeDown := chainlinkNodes[:len(chainlinkNodes)/2+1]
+			for _, nodeToTakeDown := range firstHalfToTakeDown {
+				err := nodeToTakeDown.DeleteJob("1")
+				Expect(err).ShouldNot(HaveOccurred(), "Could not delete the job from one of the nodes")
+				err = chainClient.WaitForEvents()
+				Expect(err).ShouldNot(HaveOccurred(), "Error deleting the Keeper job from the node")
+			}
+			log.Info().Msg("Successfully managed to take down the first half of the nodes")
+
+			// Assert that upkeeps are still performed and their counters have increased
+			Eventually(func(g Gomega) {
+				for i := 0; i < len(upkeepIDs); i++ {
+					currentCounter, err := consumers[i].Counter(context.Background())
+					g.Expect(err).ShouldNot(HaveOccurred(), "Failed to retrieve consumer counter"+
+						" for upkeep at index "+strconv.Itoa(i))
+					g.Expect(currentCounter.Int64()).Should(BeNumerically(">", initialCounters[i].Int64()),
+						"Expected counter to have increased from initial value of %s, but got %s",
+						initialCounters[i], currentCounter)
+				}
+			}, "3m", "1s").Should(Succeed())
```

Comment:
nit. keep it at 1m, 1s?

Reply:
I noticed in some of the previous commits that sometimes the test was failing when we had 1m here. This happens because the servicing is slower once we take down half of the nodes. I could put 2m? And if it's flaky in the future we can expand it back to 3.

Reply:
ah makes sense. fine with either